### PR TITLE
[14.0] Add purchase_partner_incoterm

### PR DESCRIPTION
**Pull request by sonhd91**

### CODE REVIEW

File: purchase_incoterm/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+{
+    'name': 'Purchase Incoterm',
```
Comment (sebalix) [inline purchase_incoterm/__manifest__.py]: I would rename the addon `purchase_partner_incoterm` to be inline with the existing [`sale_partner_incoterm`](https://github.com/OCA/sale-workflow/tree/14.0/sale_partner_incoterm).
+ Update the descriptions/user labels as well to be coherent:
```suggestion
    "name": "Default purchase incoterm per partner",
    "summary": "Set the supplier preferred incoterm on each purchase order",
```

File: purchase_incoterm/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    'summary': 'Add a an incoterm field for supplier and use it on purchase order',
+    'version': '14.0.1.0.0',
+    'category': 'Purchase',
+    'author': 'Camptocamp, Odoo Community Association (OCA)',
+    'license': 'AGPL-3',
+    'depends': [
+        'account',
+        'base',
```
Comment (sebalix) [inline purchase_incoterm/__manifest__.py]:
```suggestion
```

File: purchase_incoterm/models/purchase_order.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    @api.onchange('partner_id')
+    def _onchange_partner(self):
+        if self.partner_id:
+            self.incoterm_id = self.partner_id.purchase_incoterm_id
+        else:
+            self.incoterm_id = False
```
Comment (sebalix) [inline purchase_incoterm/models/purchase_order.py]: There is already a `onchange_partner_id` on `purchase.order`, better to re-use it and simplify the implementation:
```suggestion
    def onchange_partner_id(self):
        res = super().onchange_partner_id()
        self.incoterm_id = self.partner_id.purchase_incoterm_id
        return res
```

File: purchase_incoterm/models/res_partner.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+from odoo import fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    purchase_incoterm_id = fields.Many2one(
+        comodel_name="account.incoterms",
+        ondelete="restrict",
+        string="Purchase Incoterm",
```
Comment (sebalix) [inline purchase_incoterm/models/res_partner.py]: Use the same terms than `sale_partner_incoterm`:
```suggestion
        string="Default Purchase Incoterm",
        help="The default incoterm for new purchase orders for this supplier.",
```

**Review by sebalix — CHANGES_REQUESTED** [review]

File: purchase_incoterm/models/res_partner.py (hunk 4 above)
Comment (sonhd91) [reply purchase_incoterm/models/res_partner.py]: Thanks for reivew, branch updated

File: purchase_incoterm/models/purchase_order.py (hunk 3 above)
Comment (sonhd91) [reply purchase_incoterm/models/purchase_order.py]: Thanks for reivew, branch updated

File: purchase_incoterm/__manifest__.py (hunk 1 above)
Comment (sonhd91) [reply purchase_incoterm/__manifest__.py]: Thanks for reivew, branch updated

Comment (TDu) [reply purchase_incoterm/__manifest__.py]: I wonder it the name in the manifest should not be the name of the module without the underscore ?

**Comment by TDu** [comment]
Renaming the module as proposed by @sebalix into `purchase_partner_incoterm` seems like a good idea

**Comment by sebalix** [comment]
@sonhd91 could you rename the module `purchase_partner_incoterm`?

**Comment by sonhd91** [comment]
> @sonhd91 could you rename the module `purchase_partner_incoterm`?

module name changed, thanks

**Review by TDu — APPROVED** [review]

**Review by sebalix — APPROVED** [review]

**Pull request merged**